### ISSUE TITLE
[REVIEW]: The AutoActive Research Environment

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#4061]
**Submitting author:** @kasbot (<a href="http://orcid.org/0000-0002-9983-0041">Kasper Bøtker Rasmussen</a>)
**Repository:** https://github.com/SINTEF/AutoActive-ActivityPresenter.git
**Branch with paper.md** (empty if default branch): 
**Version:** v1.1.1
**Editor:** @prashjha
**Reviewers:** @AustinTSchaffer, @erik-whiting
**Archive:** 10.5281/zenodo.6393705

 

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/7194cbb9c3d2f4ef817780d686cfaf1a/status.svg)](https://joss.theoj.org/papers/7194cbb9c3d2f4ef817780d686cfaf1a)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/7194cbb9c3d2f4ef817780d686cfaf1a"><img src="https://joss.theoj.org/papers/7194cbb9c3d2f4ef817780d686cfaf1a/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/7194cbb9c3d2f4ef817780d686cfaf1a/status.svg)](https://joss.theoj.org/papers/7194cbb9c3d2f4ef817780d686cfaf1a)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/SINTEF/AutoActive-ActivityPresenter.git" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@AustinTSchaffer & @erik-whiting, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @prashjha know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

## Review checklist for @AustinTSchaffer
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/SINTEF/AutoActive-ActivityPresenter.git">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/SINTEF/AutoActive-ActivityPresenter.git">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@kasbot) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] #4095
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

## Review checklist for @erik-whiting
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/SINTEF/AutoActive-ActivityPresenter.git">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/SINTEF/AutoActive-ActivityPresenter.git">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@kasbot) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @AustinTSchaffer, @erik-whiting it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
Wordcount for `paper.md` is 1459

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.21105/joss.02431 is OK
- 10.1109/ICHI48887.2020.9374371 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.26 s (816.9 files/s, 143778.7 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
C#                             154           6446           4141          23953
MSBuild script                  18             57              7           1169
XAML                            24             62              1           1072
XML                             13             26             45            231
Markdown                         2             41              0            213
JSON                             1              0              0            117
Python                           1             18              5             32
TeX                              1              2              0             27
-------------------------------------------------------------------------------
SUM:                           214           6652           4199          26814
-------------------------------------------------------------------------------


Statistical information for the repository 'e434e72af70f9ff39311a8c4' was
gathered on 2022/01/13.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Sigurd M. Albrektsen             1            55              0          100.00

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Sigurd M. Albrektsen         55          100.0          0.0                5.45
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.04061/joss.04061/10.21105.joss.04061.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.04061/joss.04061/10.21105.joss.04061.pdf) :page_facing_up: :point_left:

**Comment by prashjha** [comment]
Thank you, @AustinTSchaffer and @erik-whiting, for agreeing to be reviewers. Please read the first couple of comments in this thread and also reviewer guidelines [here](https://joss.readthedocs.io/en/latest/reviewer_guidelines.html). Also, you can browse the [closed "REVIEW" issues](https://github.com/openjournals/joss-reviews/issues?q=label%3Areview+is%3Aclosed) on the "joss-reviews" repository to get some ideas on how to complete the reviews. Good luck!

**Comment by prashjha** [comment]
@erik-whiting, I will suggest opening an issue to rename the file as just LICENSE as it can get confusing.

**Comment by erik-whiting** [comment]
@prashjha another question for you: Regarding the documentation checks, there's a checkbox item that says
> Functionality documentation: Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?

Most of the core functionality is well documented and It doesn't seem like there's an API for this, but there are MATLAB and Python libraries that are supposed to work with this submission's AAZ file format. These libraries seem like they're a separate from the submission, but the Python library doesn't appear to have great documentation. Is this part of "API method documentation" and should I advise the authors to provide some documentation for this (and possibly the MATLAB library too) or are these things not technically part of the submission and the existing documentation is good enough?

**Comment by erik-whiting** [comment]
@prashjha bumping my question above

**Comment by prashjha** [comment]
@erik-whiting, sorry I missed it. I am not 100% sure about this. If you think those scripts in Matlab/python are something users will need to use at some point in testing/using this library then the scripts should be well documented. I would consider the example or demo files integral to the library as these are the files any user needs to run first. If such files are too many, we of course can expect good documentation for all of them.

**Comment by kasbot** [comment]
@prashjha , Should we do any of these changes now? or should we wait untill both reviewers are completly finished?

**Comment by erik-whiting** [comment]
> @prashjha , Should we do any of these changes now? or should we wait untill both reviewers are completly finished?

@kasbot I'm not the editor but I think I can answer this for you. usually, once the issues are addressed, we just mark those check boxes as done. it's up to you when you want to start addressing issues but you don't have to wait for the reviews to finish.

**Comment by prashjha** [comment]
@erik-whiting, thanks. That is exactly what I would suggest.

**Comment by AustinTSchaffer** [comment]
@kasbot I left 2 issue threads on the repository about clarifying the installation steps, OS-compatibility, and community guidelines. I believe this submission is overall a high quality project, apart from a few minor clarifications that could be added to the readme. I love that the repository includes links to videos that show how the software works and what it does, which was helpful in determining that the software does work and accomplishes what the paper and readme declare.

I'm personally OK with having a non-standard license file name, though the license file is not complete and contains template placeholders. I created another issue thread for that.

**Comment by whedon** [comment]
:wave: @AustinTSchaffer, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @erik-whiting, please update us on how your review is going (this is an automated reminder).

**Comment by erik-whiting** [comment]
> 👋 @erik-whiting, please update us on how your review is going (this is an automated reminder).

I opened two issues for the general checks and paper review. I have not had time to start the other reviews, but it's on my agenda, I promise 👍

**Comment by kasbot** [comment]
@prashjha, just letting you know that we have made changes according to comments from reviewers. All changes are made in the branch called joss_review_round_1, should I wait to merge the changes untill the reviewers have finished?

**Comment by prashjha** [comment]
@kasbot if you want you can merge the changes!

**Comment by AustinTSchaffer** [comment]
@kasbot I looked through your `joss_review_round_1` branch, looks good! I'll close my issue threads on your repository once the branch is merged, and check off the rest of my checklist above.

**Comment by kasbot** [comment]
@AustinTSchaffer , @prashjha and @erik-whiting  the changes have now been merged into the develope branch

**Comment by erik-whiting** [comment]
Update on my end: I think @kasbot 's changes are good so far, I will test out the functionality and note any performance improvements later this weekend hopefully. Thanks everyone for your patience so far.

**Comment by AustinTSchaffer** [comment]
@kasbot @prashjha Everything looks good on my end. I have completed the review checklist.

**Comment by prashjha** [comment]
@AustinTSchaffer thanks for taking the time to complete the review.

**Comment by prashjha** [comment]
@erik-whiting thank you for the update and sounds good. Take your time.

**Comment by erik-whiting** [comment]
@prashjha hey Prashant, thanks for the comment but I just finished my review. I've completed the checklist and I think this is ready for approval. The authors did some really cool stuff with this project, I'm glad I got to be part of it.

Thanks @kasbot

**Comment by erik-whiting** [comment]
Hey @prashjha just wanted to ping you and let you know I'm pretty sure this review is complete

**Comment by prashjha** [comment]
@editorialbot generate pdf

**Comment by prashjha** [comment]
@editorialbot commands

**Comment by prashjha** [comment]
@erik-whiting thank you for your efforts and sorry it took a long time to reply! I was on vacation and am now back. Yes, the review is complete. @kasbot, I am reading the draft and may have a few suggestions. Once you incorporate (or not!) them, I will go ahead and hand your submission to the current editor-in-chief for final acceptance.

**Comment by prashjha** [comment]
Hi @kasbot, see if these suggestions make sense for your draft:

- Line 24: should you use "optimization" here?
- Line 24: in the sentence "... and evaluation of algorithms to extract high-level, quantified analysis from sensor data", it seems there is something missing between "high-level" and "quantified" or maybe comma is inappropriate here?
- Line 27: sentence has many errors. E.g., it should be "consists". Further, you need to use commas and "and" to list three modules, for example, "ActivityPresenter, the accompanying Matlab, and Python toolboxes". Moreover, you say these three are open-source modules but Matlab is not open-source. Did you mean Matlab script that can be run using Octave? Also, "the accompanying Matlab" is meaningless. Did you mean to say the accompanying Matlab scripts? With little effort this sentence can be greatly improved.
- Line 64: "... sessions is a key to ..." and "store data in ARE; they are not ...".
- Line 67: "When a session is saved, it becomes ...".
- Line 68: "This allows sessions ...".
 

After you have updated the draft, can you also do (if not done already) a 'tagged' release of your code
archive the release using zenedo or other methods? Make sure that the title of zenedo archive matches with the title of this JOSS submission.

Once you are done, I will hand your paper to EiC for the final decision.

**Comment by kasbot** [comment]
@editorialbot generate pdf

**Comment by kasbot** [comment]
@whedon generate pdf

**Comment by kasbot** [comment]
Hi @prashjha,
Thank you for your suggestions. The draft has now been updated according to your suggestions. We have also done a tagged release of our code using zenodo which can be found here https://zenodo.org/search?page=1&size=20&q=AutoActive

**Comment by kasbot** [comment]
Hi @prashjha, I am sorry for the late respons, I am on paternity leave right now, so I am doing my best keeping up with everything work related. Please use the first one (https://zenodo.org/record/6393705#.Yk86bShBxPY)

**Comment by prashjha** [comment]
@editorialbot set 10.5281/zenodo.6393705 as archive

**Comment by prashjha** [comment]
@editorialbot recommend-accept

**Comment by prashjha** [comment]
@kasbot congrats! and no worries. I have handed your paper to EiC for final decision. Congrats again!!

@AustinTSchaffer, @erik-whiting, thank you both for your effort in reviewing this submission. It is very much appreciated. :)

**Comment by erik-whiting** [comment]
happy to help @prashjha. Nice work @kasbot !

**Comment by arfon** [comment]
@kasbot – I made a few minor edits to the paper here: https://github.com/SINTEF/AutoActive-ActivityPresenter/pull/15

**Comment by kasbot** [comment]
Thank you @arfon. I have merged your paper fixes, please let me know if there is anything else I need to do

**Comment by arfon** [comment]
@editorialbot accept

**Comment by arfon** [comment]
@AustinTSchaffer, @erik-whiting – many thanks for your reviews here and to @prashjha for editing this submission! JOSS relies upon the volunteer effort of people like you and we simply wouldn't be able to do this without you ✨ 

@kasbot – your paper is now accepted and published in JOSS :zap::rocket::boom:

**Comment by aliverud** [comment]
Thanks a lot for the review and publishing!

There is a broken link for the author "Anders E. Liverud" at the publication page. Could you please correct? (the ".E" is missing in the authors page: https://joss.theoj.org/papers/by/Anders%20Liverud vs. https://joss.theoj.org/papers/by/Anders%20E.%20Liverud)
